File: backend/app/schema/documents.py
from datetime import datetime

from pydantic import BaseModel


class DocumentResponse(BaseModel):

    id: int

    filename: str

    file_type: str

    file_size: int

    processing_status: str

    storage_path: str

    file_url: str

    created_at: datetime

    class Config:
        from_attributes = True


class UploadResponse(BaseModel):

    message: str

    documents: list[DocumentResponse]

File: frontend/types/documents.ts
export interface RecentDocument {
  id: number;
  filename: string;
  processing_status: string;
  created_at: string;
}

export interface DocumentAsset {
  id: number;
  filename: string;
  file_type: string;
  file_size: number;
  processing_status: string;
  uploaded_by: string;
  created_at: string;
}
